crates/bevy_ui/src/flex/mod.rs

```diff
@@ -262,10 +270,13 @@ pub fn flex_node_system(
         flex_surface.set_window_children(primary_window.id(), root_node_query.iter());
     }
 
-    // update children
+    // update and remove children
     for (entity, children) in &children_query {
         flex_surface.update_children(entity, children);
     }
+    for entity in &removed_children {
+        flex_surface.try_remove_children(entity);
+    }
```

Comment:
We should handle removals before the children query in case the children component was removed and then re-added in that frame.